costmap_2d/src/layered_costmap.cpp

```diff
@@ -82,83 +104,190 @@ LayeredCostmap::~LayeredCostmap()
 void LayeredCostmap::resizeMap(unsigned int size_x, unsigned int size_y, double resolution, double origin_x,
                                double origin_y, bool size_locked)
 {
-  boost::unique_lock<Costmap2D::mutex_t> lock(*(costmap_.getMutex()));
+  boost::unique_lock<Costmap2D::mutex_t> lock(*(static_costmap_.getMutex()));
   size_locked_ = size_locked;
-  costmap_.resizeMap(size_x, size_y, resolution, origin_x, origin_y);
+  static_costmap_.resizeMap(size_x, size_y, resolution, origin_x, origin_y);
+  for(Costmap2D& costmap : timed_costmaps_)
+  {
+    boost::unique_lock<Costmap2D::mutex_t> lock(*(costmap.getMutex()));
```

Comment:
S: rather than re-obtaining for every element, acquire once outside the loop:
```c++
{
    boost::unique_lock<Costmap2D::mutex_t> lock(*(costmap.getMutex()));
    for(Costmap2D& costmap : timed_costmaps_)
    {
       costmap.resizeMap(size_x, size_y, resolution, origin_x, origin_y);
    }
}
```

Reply:
don't we need to lock each individual costmap in the timed_costmap vector?

Reply:
oh yes, my bad